packages/main/src/themes/SegmentedButton.css

```diff
@@ -3,10 +3,13 @@
 
 :host(:not([hidden])) {
 	display: inline-block;
+	min-width: calc(var(--_ui5_segmented_btn_items_count) * 2.5rem);
 }
 
 .ui5-segmented-button-root {
-	display: flex;
+	width: inherit;
+	display: grid;
```

Comment:
This is great reduce of JS, thank you @nnaydenow

Reply:
Absolutely, thank you a bunch @nnaydenow for the proposed solution!